app/models/piece.rb

```diff
@@ -1,3 +1,81 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
+
+  def check_square(x, y)
+    return true if @game.pieces.where(row_coordinate: x, column_coordinate: y).any?
+  end
+
+  def check_horizontal(s_x, _d_x, s_y, d_y)
+    strt = [s_y, d_y].min + 1
+    fin = [s_y, d_y].max - 1
+
+    while strt <= fin
+      return true if check_square(s_x, strt)
+      strt += 1
+    end
+  end
+
+  def check_vertical(s_x, d_x, s_y, _d_y)
+    strt = [s_x, d_x].min + 1
+    fin = [s_x, d_x].max - 1
+
+    while strt <= fin
+      return true if check_square(strt, s_y)
+      strt += 1
+    end
+  end
+
+  def check_diagonal(s_x, d_x, s_y, d_y)
+    slope = (d_y - s_y) / (d_x - s_x)
+    if slope > 0
+      start_x = [s_x, d_x].min + 1
+      end_x = [s_x, d_x].max + 1
+      start_y = [s_y, d_y].min - 1
+      while start_x <= end_x
+        return true if check_square(start_x, start_y)
+        start_x += 1
+        start_y += 1
+      end
+    else
+      start_x = [s_x, d_x].min + 1
+      end_x = [s_x, d_x].max - 1
+      start_y = [s_y, d_y].max - 1
+      while start_x <= end_x
+        return true if check_square(start_x, start_y)
+        start_x += 1
+        start_y -= 1
+      end
+    end
+  end
+
+  def obstructed?(d_x, d_y)
+    @game = Game.find(game_id)
```

Comment:
since we're currently inside a `Piece` and it has `belongs_to :game`. You can use `game` anywhere within `Piece` and Rails will do the find/load the game for you.